kyo-core/shared/src/main/scala/kyo/Channel.scala

```diff
@@ -158,6 +158,57 @@ object Channel:
           */
         def full(using Frame): Boolean < (Abort[Closed] & IO) = IO.Unsafe(Abort.get(self.full()))
 
+        private def emitChunks(maxChunkSize: Int = Int.MaxValue)(
+            using
+            Tag[Emit[Chunk[A]]],
+            Frame
+        ): Ack < (Emit[Chunk[A]] & Abort[Closed] & Async) =
+            if maxChunkSize <= 0 then Ack.Stop
+            else if maxChunkSize == 1 then
+                Loop(()): _ =>
+                    Channel.take(self).map: v =>
+                        Emit.andMap(Chunk(v)):
+                            case Ack.Stop => Loop.done(Ack.Stop)
+                            case _        => Loop.continue(())
+            else
+                val drainEffect =
+                    if maxChunkSize == Int.MaxValue then Channel.drain(self)
+                    else Channel.drainUpTo(self)(maxChunkSize - 1)
+                Loop[Unit, Ack, Abort[Closed] & Async & Emit[Chunk[A]]](()): _ =>
+                    Channel.take(self).map: a =>
+                        drainEffect.map: chunk =>
+                            Emit.andMap(Chunk(a).concat(chunk)):
+                                case Ack.Stop => Loop.done(Ack.Stop)
+                                case _        => Loop.continue(())
```

Comment:
We should make a version of the loop `Loop.stateless`

Reply:
Agreed! It's a common pattern